### ISSUE TITLE
Crashes service is not started on iOS when managed stripping level is set to medium or high

### ISSUE THREAD

**Issue by KilledWhale** [Microsoft/AppCenter-SDK-Unity#386]
### **Description**
Crashes service is not started on iOS when managed stripping level is set to medium or high.
The reason for this is that as part of managed bytecode stripping (See [https://docs.unity3d.com/2019.3/Documentation/Manual/ManagedCodeStripping.html](url)) UnityLinker determines that `Crashes.StartCrashes` is not called from anywhere and can be safely removed.

### **Repro Steps**
1. Create new Unity project and add and configure App Center SDK + Crashes package to the project
2. Go to iOS player settings and change _Managed Stripping Level_ to _Medium_
3. Create and run iOS build and observe crashes service not starting

### **How to Fix This**
Unity provides two ways (link.xml files or Preserve attribute) to instruct the UnityLinker that something is accessed by reflection and should not be stripped away. See [https://docs.unity3d.com/2019.3/Documentation/Manual/ManagedCodeStripping.html](url).

Since the SDK is not using assembly definitions, preserving the whole assembly is probably not a good idea and the least intrusive solution would be to add `[UnityEngine.Scripting.Preserve]` attribute to `Crashes.StartCrashes` -method.

Looking at the _AppCenterBehavior.cs_ (see following snippet), it seems likely that on iOS the Push service is affected as well and `Push.StartPush` may require similar treatment.

```CSharp
#if UNITY_IOS || UNITY_ANDROID
        else
        {
            foreach (var service in services)
            {
#if UNITY_IOS || UNITY_ANDROID
                // On iOS and Android we start crash service here, to give app an opportunity to assign handlers after crash and restart in Awake method
                var startCrashes = service.GetMethod("StartCrashes");
                if (startCrashes != null)
                    startCrashes.Invoke(null, null);
#endif
#if UNITY_IOS
                var startPush = service.GetMethod("StartPush");
                if (startPush != null)
                    startPush.Invoke(null, null);
#endif
            }
        }
#endif
```

### **Details**
1. Which SDK version are you using?
    - 2.5.1
2. Which platform did you experience the issue on?
    - iOS 13
    - Android is likely affected as well when IL2CPP and managed bytecode stripping are being used.
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - iPhone 11

![settings](https://user-images.githubusercontent.com/196424/70970164-53c06880-20a6-11ea-9fe9-c636e2783114.png)

**Comment by Jamminroot** [comment]
Hey, thanks for bringing this to our attention!
We'll have a look at this.

**Comment by AnastasiaKubova** [comment]
The fix had been released in [3.1.0](https://github.com/microsoft/appcenter-sdk-unity/releases/tag/3.1.0).
